lib/anonymizer/lib/anonymizer/column.rb

```diff
@@ -0,0 +1,69 @@
+module Anonymizer
+  class Column
+    attr_reader :column, :table_name, :scope
+
+    def initialize(table_name, column, scope: nil)
+      @table_name = table_name
+      @column = column
+      @scope = scope
+    end
+
+    def anonymize!
+      if column.type.in?(%i[string text]) && column.null
+        # Pour limiter la confusion lors de l'exploitation des données, on transforme les chaines vides en null
+        blank_value = column.array ? "{}" : ""
+        update where: arel_table[column.name].eq(blank_value), value: nil
+      end
+
+      update where: arel_table[column.name].not_eq(nil), value: anonymous_value
+    end
+
+    private
+
+    def column_name = column.name
+
+    def arel_table
+      @arel_table ||= Arel::Table.new(table_name)
+    end
+
+    def update(where:, value:)
+      ActiveRecord::Base.connection.execute(
+        Arel::UpdateManager
+          .new(arel_table)
+          .where(scope ? scope.and(where) : where)
```

Comment:
par ex : 
- `scope` = receipts créés il y a plus de 2 ans
- `where` = column is null

je ne sais pas trop comment clarifier ça dans ce code 🤷